### PR TITLE
python312Packages.pyrdfa3: remove imp, python311Packages.extruct: 0.13.0 -> 0.16.0, python312Packages.mf2py: 1.1.3 -> 2.0.1 

### PR DESCRIPTION
**Pull request by fabaff**
## Description of changes
Fix build (https://hydra.nixos.org/build/246437176)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>python311Packages.extruct</li>
    <li>python311Packages.extruct.dist</li>
    <li>python311Packages.mf2py</li>
    <li>python311Packages.mf2py.dist</li>
    <li>python311Packages.pyrdfa3</li>
    <li>python311Packages.pyrdfa3.dist</li>
    <li>python311Packages.recipe-scrapers</li>
    <li>python311Packages.recipe-scrapers.dist</li>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.mf2py</li>
    <li>python312Packages.mf2py.dist</li>
    <li>python312Packages.pyrdfa3</li>
    <li>python312Packages.pyrdfa3.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyrdfa3/default.nix [hunk 1]
```diff
@@ -1,51 +1,48 @@
...
   };
 
-  patches = [
-    (fetchpatch {
-      name = "CVE-2022-4396.patch";
-      url = "https://github.com/RDFLib/pyrdfa3/commit/ffd1d62dd50d5f4190013b39cedcdfbd81f3ce3e.patch";
-      hash = "sha256-prRrOwylYcEqKLr/8LIpyJ5Yyt+6+HTUqH5sQXU8tqc=";
-    })
-  ];
-
```
Comment (ambroisie) [inline pkgs/development/python-modules/pyrdfa3/default.nix:29]: Why remove the CVE patch?

Comment (fabaff) [reply pkgs/development/python-modules/pyrdfa3/default.nix]: Reverted. The patch should stay as there was no new release published which contain the change.

Thanks for bringing this up..

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tandoor-recipes</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>python311Packages.extruct</li>
    <li>python311Packages.extruct.dist</li>
    <li>python311Packages.mf2py</li>
    <li>python311Packages.mf2py.dist</li>
    <li>python311Packages.pyrdfa3</li>
    <li>python311Packages.pyrdfa3.dist</li>
    <li>python311Packages.recipe-scrapers</li>
    <li>python311Packages.recipe-scrapers.dist</li>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.mf2py</li>
    <li>python312Packages.mf2py.dist</li>
    <li>python312Packages.pyrdfa3</li>
    <li>python312Packages.pyrdfa3.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
  </ul>
</details>

**Comment by ambroisie** [comment]
@fabaff I would like to investigate the `tandoor-recipes` failure before approving.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 281730` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.extruct</li>
    <li>python312Packages.extruct.dist</li>
    <li>python312Packages.mf2py</li>
    <li>python312Packages.mf2py.dist</li>
    <li>python312Packages.pyrdfa3</li>
    <li>python312Packages.pyrdfa3.dist</li>
    <li>python312Packages.recipe-scrapers</li>
    <li>python312Packages.recipe-scrapers.dist</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python311Packages.extruct</li>
    <li>python311Packages.extruct.dist</li>
    <li>python311Packages.mf2py</li>
    <li>python311Packages.mf2py.dist</li>
    <li>python311Packages.pyrdfa3</li>
    <li>python311Packages.pyrdfa3.dist</li>
    <li>python311Packages.recipe-scrapers</li>
    <li>python311Packages.recipe-scrapers.dist</li>
    <li>tandoor-recipes</li>
  </ul>
</details>

**Comment by fabaff** [comment]
`tandoor-recipes` should be good for now.

**Pull request merged**